sparse/core.py

```diff
@@ -328,6 +328,11 @@ def T(self):
     def dot(self, other):
         return dot(self, other)
 
+    __matmul__ = dot
+
+    def __rmatmul__(self, other):
+        return dot(np.array(other), self)
```

Comment:
Hrm, this coercion is a bit concerning.  What if, for example, `other` is a scipy.sparse matrix?  Coercing it to a numpy array may be unwelcome.